src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -51,9 +57,14 @@ public CreatedWorkspace createWorkspace(
     return createJob.submitAndWait(CreatedWorkspace.class);
   }
 
+  @NewSpan
   public WorkspaceDescription getWorkspace(String id, AuthenticatedUserRequest userReq) {
-
-    samService.workspaceAuthz(userReq, id, SamUtils.SAM_WORKSPACE_READ_ACTION);
+    Span newSpan = tracer.nextSpan().name("workspaceAuthz");
+    try (Tracer.SpanInScope ws = tracer.withSpanInScope(newSpan.start())) {
+      samService.workspaceAuthz(userReq, id, SamUtils.SAM_WORKSPACE_READ_ACTION);
+    } finally {
+      newSpan.finish();
+    }
```

Comment:
Here's an example trace 

- We have @NewSpan, which creates a span with the same name as the method
- We are also creating a span explicitly, and then closing it. I tried to make this cleaner by attempting to make a Span class that implements Autocloseable, but it got a little hairy.

I imagine we wouldn't make too many spans explicitly, and use @NewSpan generally

Reply:
Are these two different ways of opening spans, or do we need to use both in some places?

Reply:
Yeah, I'm just modeling two different ways of creating a span. I think we'll probably use @NewSpan generally (which, btw, you can also specify a name instead of automatically using the name of the method by doing @NewSpan("different name") ).

Reply:
For the explicitly created span, `SpanInScope` implements `Closeable`, not `AutoCloseable`, which is why we need to explicitly close the span in that `finally { ... }` block :( 
I attempted to try to finagle trying to make something that uses `Autocloseable` but it got messy and is probably not worth it

Reply:
The explicitly-created spans are definitely more verbose/awkward than the opencensus implementation. I *do* think we need the level of detail of multiple inner spans, inside a complicated API. Is your thinking here that if we need inner spans, all the business logic for that span should be refactored into a standalone method, which could use @NewSpan?

Reply:
Yes - I was thinking we could generally break out methods if we need to create spans for blocks of code and use @NewSpan.

But, idk if that's good practice. It keeps the code cleaner. But then we're restructuring our code for the sake of a trace. Do you think it's worth it?